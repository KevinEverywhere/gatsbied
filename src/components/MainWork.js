import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'gatsby'
import WorkItem from './WorkItem'
import RecommendationItem from './RecommendationItem'

class MainWork extends React.Component {
  constructor(props){
    super();
  }
  drawrecommendations() {
    return this.props.recommendations.map((item, i) => {
       return(
         <RecommendationItem key={i} item={item} />
       )
     });
  }
  drawjobs() {
    return this.props.jobs.map((item, i) => {
       return(
         <WorkItem key={i} item={item} />
       )
     });
   }
   oldrender() {
     let str;
     if(this.props[this.props.isSelected]){
       str=this[`draw${this.props.isSelected}`]();
     }else{
       str='';
     }
     return str;
  }
  render() {
    const minWidth = 240;
    return (
      <main>
        <div style={{
          textShadow:`1px 1px black`,
          padding: '1rem',
          color:'white',
          fontWeight: 'bold',
          width: '60%',
          minWidth: `${minWidth}px`
        }}>
          <h2>About Kevin</h2>
          <p>Kevin Ready has kept up with the most current technolgies creating code, players, templates and content for startups, agencies, and digital media clients, since Bill Clinton was in office. This site was created in 2019 using react and gatsby, with graphql, aframe, vanilla JavaScript, and others, optimized for mobile. Kevin is often looking for contract or permanent work using software and technologies for which he is passionate.</p>
          <p style={{display: 'none'}}>
          More information can be found at the following links.
          </p>
        </div>
        <ul className='horizontalBar'>
          <li><Link to='/contact'>Contact</Link></li>
          <li><a href='https://linkedin.com/in/planetkevin/'>LinkedIn</a></li>
          <li><a href='https://github.com/kevineverywhere/'>GitHub</a></li>
        </ul>
      </main>
    )
  }
}

MainWork.propTypes = {
  jobs: PropTypes.arrayOf(
    PropTypes.shape({
      company: PropTypes.string,
      title: PropTypes.string,
      description: PropTypes.string,
      location: PropTypes.string,
    })
  )
}

export default MainWork


// started: PropTypes.string,
// finished: PropTypes.string,
