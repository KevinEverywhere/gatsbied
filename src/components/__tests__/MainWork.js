import React from "react"
import renderer from "react-test-renderer"
import MainWork from "../MainWork"

describe("MainWork", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<MainWork />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
