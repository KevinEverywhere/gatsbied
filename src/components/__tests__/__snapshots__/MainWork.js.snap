// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainWork renders correctly 1`] = `
<main>
  <div
    style={
      Object {
        "color": "white",
        "fontWeight": "500",
        "minWidth": "240px",
        "padding": "1rem",
        "textShadow": "1px 1px black",
        "width": "60%",
      }
    }
  >
    <h2>
      About Kevin
    </h2>
    <p>
      Kevin Ready has kept up with the most current technologies creating code, players, templates and content for startups, agencies, and digital media clients, since Bill Clinton was in office. This site was created in 2019 using react and gatsby, with graphql, aframe, sass, vanilla JavaScript, and others, optimized for mobile. He is looking for contract or permanent work using software and technologies for which he is passionate.
    </p>
    <p
      style={
        Object {
          "display": "none",
        }
      }
    >
      More information can be found at the following links.
    </p>
  </div>
  <ul
    className="horizontalBar"
  >
    <li>
      <a
        href="/contact"
      >
        Contact
      </a>
    </li>
    <li>
      <a
        href="https://linkedin.com/in/planetkevin/"
      >
        LinkedIn
      </a>
    </li>
    <li>
      <a
        href="https://github.com/kevineverywhere/"
      >
        GitHub
      </a>
    </li>
  </ul>
</main>
`;
